diff --git a/app/server/ruby/lib/sonicpi/scsynthexternal.rb b/app/server/ruby/lib/sonicpi/scsynthexternal.rb
index cac5edddb..3806351f4 100644
--- a/app/server/ruby/lib/sonicpi/scsynthexternal.rb
+++ b/app/server/ruby/lib/sonicpi/scsynthexternal.rb
@@ -423,7 +423,7 @@ module SonicPi
         "-z" =>  block_size.to_s,
         "-i" => "2",
         "-o" => "2",
-        "-U" => "/usr/lib/SuperCollider/plugins"}
+        "-U" => "/app/lib/SuperCollider/plugins"}
 
       scsynth_opts = @default_scsynth_opts.merge(local_scsynth_opts).merge(@user_scsynth_opts)
 
